### PR TITLE
Type hints

### LINKED ISSUE #250: Use Type Hinting in metacatalog
We should implement type hinting in metacatalog, as this simplifies further development.

We can start with the following list, to which we can add at any point.

**Implement Type Hinting:**
- [ ] Models
- [ ] API

### PR DESCRIPTION
**Pull request by mmaelicke**
closes #250 

This is still draft. Not all API and model files are yet type hinted.

### CODE REVIEW

File: metacatalog/models/entry.py [hunk 1]
```diff
@@ -615,7 +616,7 @@ def details_table(self, fmt='html'):
         else:
             raise ValueError("fmt has to be in ['html', 'latex', 'markdown']")
 
-    def add_details(self, details=None, commit=False, **kwargs):
+    def add_details(self, details=None, commit: bool = False, **kwargs):
```
Comment (AlexDo1) [inline metacatalog/models/entry.py:619]: I made the same comment before, would'n we give a typehint for return `None` here?  
`-> None`

File: metacatalog/models/entry.py [hunk 2]
```diff
@@ -683,7 +684,7 @@ def add_details(self, details=None, commit=False, **kwargs):
                 session.rollback()
                 raise e
 
-    def export(self, path=None, fmt='JSON', **kwargs):
+    def export(self, path: str = None, fmt: str = 'JSON', **kwargs):
```
Comment (AlexDo1) [inline metacatalog/models/entry.py:687]: `-> None` ?

**Review by AlexDo1 — CHANGES_REQUESTED** [review]
That was a lot of code to read, I just requested some small changes.  

I really like the typehints, this will help a lot with further development

File: metacatalog/api/catalog.py [hunk 3]
```diff
@@ -4,18 +4,24 @@
 specific API action or model
 
 """
-from typing import Union
-
-from sqlalchemy.orm import Session
+from __future__ import annotations
```
[the remark this answers is not in the document]
Comment (mmaelicke) [reply metacatalog/api/catalog.py:7]: Yes, exactly.

File: metacatalog/models/entry.py (hunk 2 above)
Comment (mmaelicke) [reply metacatalog/models/entry.py:687]: Nope. It does return something, but I have no idea how to pipe the correct types through to this function.
I guess we keep it like it is, or do that later

**Comment by mmaelicke** [comment]
Thanks @AlexDo1! You can review again.

**Review by AlexDo1 — APPROVED** [review]
Looks good!

**Pull request merged**